reportng/src/java/main/org/uncommons/reportng/ReportNGUtils.java

```diff
@@ -87,8 +65,25 @@ public String formatDuration(long startMillis, long endMillis)
 
     public String formatDuration(long elapsed)
     {
+
```

Comment:
I'd recommend using the java TimeUnit package for this (java.util.concurrent.TimeUnit)

```java
private static String formatInterval(final long l) {
   final long hr = TimeUnit.MILLISECONDS.toHours(l);
   final long min = TimeUnit.MILLISECONDS.toMinutes(l - TimeUnit.HOURS.toMillis(hr));
   final long sec = TimeUnit.MILLISECONDS.toSeconds(l - TimeUnit.HOURS.toMillis(hr) - TimeUnit.MINUTES.toMillis(min));
   final long ms = TimeUnit.MILLISECONDS.toMillis(l - TimeUnit.HOURS.toMillis(hr) - TimeUnit.MINUTES.toMillis(min) - TimeUnit.SECONDS.toMillis(sec));
   return String.format("%02d:%02d:%02d.%03d", hr, min, sec, ms);
}
```